### PR TITLE
Gherkin and automation of dev usability improvements

### PR DESCRIPTION
**Pull request by sanketpathak**
Epic: https://issues.redhat.com/browse/ODC-6452
Story: https://issues.redhat.com/browse/ODC-6505

Acceptance criteria:

-  Allow user to re-arrange the resources which have been added to nav by the user(This is a manual thing not included in the test)
-  Improved user experience (form based experience)
    - Form based editing of Routes
    - Form based creation and editing of Config Maps
    - Form base creation of Deployments
-  Improved discovery
    - Include Share my project on the Add page to increase discoverability
    - NS Helm Chart Repo
          1.  Add tile to Add page for discoverability
          2.  Provide a form driven creation experience
          3.  User should be able to switch back and forth from Form/YAML
          4.  change the intro text to the below & have the link in the intro text bring up the full page form
         - Browse for charts that help manage complex installations and upgrades. Cluster administrators can customize the content made available in the catalog. Alternatively, developers can try to configure their own custom Helm Chart repository
 
**Checks required for approving Epic gherkin scripts PR:**

- [x] Add @epic-number to the scenarios or feature file [e.g: @odc-xxx]
- [x] Add @to-do for automation possible scenarios
- [x] Add @regression or @smoke based on the importance of the scenario
- [x] Update the test scenarios count in [Automation status confluence Report](https://docs.jboss.org/display/ODC/Automation+Status+Report)
- [x] Check for the linter issues by executing `yarn run gherkin-lint` on frontend folder [Skip epic number tags related linter issues]

**Fix**: https://issues.redhat.com/browse/ODC-6506

**Test Setup**:

Cluster should be running on local

Check the TAGS in frontend/packages/dev-console/integration-tests/cypress.json file be: ""(@pre-condition or @smoke or @regression) and not (@manual or @to-do or @un-verified or @broken-test)"",

Command to execute:

Run a cluster locally and execute the command `yarn run dev` and then `npm run test-cypress-dev-console` simultaneously

Execute the helm-chart-repositories.feature, sharing.feature,  config-maps-dev-perspective.feature, deployment-dev-perspective.feature and route-dev-perspective.feature files

**Browser**
Chrome 99

**Test Execution Screenshot:**
Screenshot from 2022-06-09 19-23-04
Screenshot from 2022-06-09 19-20-40
Screenshot from 2022-06-09 19-19-06
Screenshot from 2022-06-09 19-17-55
Screenshot from 2022-06-09 19-38-02

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+
+        @regression
+        Scenario: Add namespaced helm chart repository using a form: A-12-TC03
+            Given user is at Helm Chart repositories page
+             When user enters Chart repository name as "helm-test1"
+              And user enters Description as "test"
+              And user enters URL as "https://raw.githubusercontent.com/IBM/charts/master/repo/community/index.yaml"
+              And user clicks on Create button
+             Then user can see "helm-test1" under Chart Repositories in Helm Charts
```
Comment (debsmita1) [inline frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature:33]:
```suggestion
             Then user can see "helm-test1" under Chart Repositories in Helm Charts catalog page
```

Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        Scenario: Configure custom Helm Chart repository.: A-12-TC03
+            Given user is at Add page
+             When user selects Helm Chart card on the Add page
+              And user clicks "try to configure their own custom Helm Chart repository" link in Helm Charts description
+             Then user can see "Create ProjectHelmChartRepository" form
+
+
+        @regression
+        Scenario: Add namespaced helm chart repository using a form: A-12-TC04
+            Given user is at Helm Chart repositories page
```
Comment (debsmita1) [inline frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature:36]:
```suggestion
            Given user is at Create ProjectHelmChartRepository page
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+             When user selects Helm Chart repositories card from Add page
+              And user switches to YAML
+             Then user can see YAML view
+
+
+        @regression
+        Scenario: Configure custom Helm Chart repository.: A-12-TC03
+            Given user is at Add page
+             When user selects Helm Chart card on the Add page
+              And user clicks "try to configure their own custom Helm Chart repository" link in Helm Charts description
```
Comment (debsmita1) [inline frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature:30]:
```suggestion
              And user clicks "try to configure their own custom Helm Chart repository" link in Helm Charts catalog description
```

Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/addFlow/helm-chart-repositories.feature]: Updated

**Comment by debsmita1** [comment]
Thanks @sanketpathak 
/lgtm

**Comment by hemantsaini-7** [comment]
/lgtm

image

image

image

image

image

File: frontend/packages/console-shared/src/components/form-utils/FormHeader.tsx [hunk 4]
```diff
@@ -18,7 +18,7 @@ const FormHeader: React.FC<FormHeaderProps> = ({ title, helpText, marginTop, mar
 
   return (
     <div style={marginStyles}>
-      <Title headingLevel="h1" size="2xl">
+      <Title headingLevel="h1" size="2xl" data-test="form title">
```
Comment (invincibleJai) [inline frontend/packages/console-shared/src/components/form-utils/FormHeader.tsx:21]: shouldn't data-test be separated `-`?

```suggestion
      <Title headingLevel="h1" size="2xl" data-test="form-title">
```

Comment (sanketpathak) [reply frontend/packages/console-shared/src/components/form-utils/FormHeader.tsx]: Updated

**Comment by sanketpathak** [comment]
Pr only affects E2E test
/label qe-approved
/label docs-approved
/label px-approved

**Pull request merged**